IDE0060 (remove unused parameter) reported for primary constructor parameters.

C:\github\roslyn\src\Workspaces\SharedUtilitiesAndExtensions\Compiler\Core\Options\OptionDefinition.cs(102,41): warning IDE0060: Remove unused parameter 'serializer' (https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0060) [C:\github\roslyn\src\CodeStyle\Core\Analyzers\Microsoft.CodeAnalysis.CodeStyle.csproj]

We need to address this as it immediately causes issues for trying to use primary constructors.

--- a/dotnet-tools.json
+++ b/dotnet-tools.json
@@ -2,7 +2,7 @@
   "isRoot": true,
   "tools": {
     "dotnet-format": {
-      "version": "6.4.326609",
+      "version": "7.0.360304",
       "commands": [
         "dotnet-format"
       ]

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -15,7 +15,7 @@
       Keep the setting conditional. The toolset sets the assembly version to 42.42.42.42 if not set explicitly.
     -->
     <AssemblyVersion Condition="'$(OfficialBuild)' == 'true' or '$(DotNetUseShippingVersions)' == 'true'">$(MajorVersion).$(MinorVersion).0.0</AssemblyVersion>
-    <MicrosoftNetCompilersToolsetVersion>4.5.0</MicrosoftNetCompilersToolsetVersion>
+    <MicrosoftNetCompilersToolsetVersion>4.6.0</MicrosoftNetCompilersToolsetVersion>
   </PropertyGroup>
   <PropertyGroup>
     <!-- Versions used by several individual references below -->
@@ -24,7 +24,7 @@
     <MicrosoftCodeAnalysisTestingVersion>1.1.2-beta1.23163.2</MicrosoftCodeAnalysisTestingVersion>
     <MicrosoftVisualStudioExtensibilityTestingVersion>0.1.149-beta</MicrosoftVisualStudioExtensibilityTestingVersion>
     <!-- CodeStyleAnalyzerVersion should we updated together with version of dotnet-format in dotnet-tools.json -->
-    <CodeStyleAnalyzerVersion>4.3.0-1.final</CodeStyleAnalyzerVersion>
+    <CodeStyleAnalyzerVersion>4.6.0</CodeStyleAnalyzerVersion>
     <VisualStudioEditorPackagesVersion>17.7.38-preview</VisualStudioEditorPackagesVersion>
     <!-- This should generally be set to $(VisualStudioEditorPackagesVersion),
          but sometimes EditorFeatures.Cocoa specifically requires a newer editor build. -->

--- a/src/Analyzers/CSharp/Tests/HideBase/HideBaseTests.cs
+++ b/src/Analyzers/CSharp/Tests/HideBase/HideBaseTests.cs
@@ -159,8 +159,8 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.HideBase
         }
 
         [Fact]
-        public async Task TestAddNewToDisorderedModifiers() =>
-            await TestInRegularAndScript1Async(
+        public async Task TestAddNewToDisorderedModifiers()
+            => await TestInRegularAndScript1Async(
                 """
                 class Application
                 {
@@ -185,8 +185,8 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.HideBase
                 """);
 
         [Fact]
-        public async Task TestAddNewToOrderedModifiersWithTrivia() =>
-            await TestInRegularAndScript1Async(
+        public async Task TestAddNewToOrderedModifiersWithTrivia()
+            => await TestInRegularAndScript1Async(
                 """
                 class Application
                 {

--- a/src/Features/Core/Portable/EditAndContinue/Utilities/Extensions.cs
+++ b/src/Features/Core/Portable/EditAndContinue/Utilities/Extensions.cs
@@ -139,11 +139,11 @@ namespace Microsoft.CodeAnalysis.EditAndContinue
             return new(
                 data.Id,
                 data.Message ?? FeaturesResources.Unknown_error_occurred,
-                updateStatus == ModuleUpdateStatus.RestartRequired ?
-                    ManagedHotReloadDiagnosticSeverity.RestartRequired :
-                    (data.Severity == DiagnosticSeverity.Error) ?
-                        ManagedHotReloadDiagnosticSeverity.Error :
-                        ManagedHotReloadDiagnosticSeverity.Warning,
+                updateStatus == ModuleUpdateStatus.RestartRequired
+                    ? ManagedHotReloadDiagnosticSeverity.RestartRequired
+                    : (data.Severity == DiagnosticSeverity.Error)
+                        ? ManagedHotReloadDiagnosticSeverity.Error
+                        : ManagedHotReloadDiagnosticSeverity.Warning,
                 fileSpan.Path ?? "",
                 fileSpan.Span.ToSourceSpan());
         }

--- a/src/VisualStudio/Core/Test.Next/Remote/SnapshotSerializationTests.cs
+++ b/src/VisualStudio/Core/Test.Next/Remote/SnapshotSerializationTests.cs
@@ -705,8 +705,8 @@ namespace Microsoft.CodeAnalysis.Remote.UnitTests
 
         private class MissingAnalyzerLoader : AnalyzerAssemblyLoader
         {
-            protected override string PreparePathToLoad(string fullPath) =>
-                throw new FileNotFoundException(fullPath);
+            protected override string PreparePathToLoad(string fullPath)
+                => throw new FileNotFoundException(fullPath);
         }
 
         private class MissingMetadataReference : PortableExecutableReference

